src/components/numbered-code-snippet/numbered-code-snippet.js

```diff
@@ -217,22 +217,81 @@ export default class NumberedCodeSnippet extends React.PureComponent {
+      /**
+       * Set the chunk equal to the contents of the line. This may include
+       * empty strings where empty lines are added to improve readability.
+       */
       const chunk = splitDisplayCode[i];
+
+      /** Set the line number. */
       const lineNumber = i + 1;
+
+      /**
+       * If there is a defined copy range that includes either the current
+       * lineNumber or an upcoming lineNumber, currentLiveRange will be equal
+       * to that copy range (an array of two numbers).
+       */
       if (currentLiveRange && lineNumber === currentLiveRange[0]) {
+        /**
+         * If there is a currentLiveRange and the lineNumber is equal to
+         * the beginning of the currentLiveRange, use endCurrentChunk to:
+         *   - stop adding lines to the current chunk
+         *   - add that chunk to the allChunks array as a non-live chunk
+         *   - start a fresh currentChunk array
+         */
         endCurrentChunk({ live: false });
       } else if (currentLiveRange && lineNumber > currentLiveRange[1]) {
+        /**
+         * If there is a currentLiveRange and the lineNumber is more than the
+         * the last line number of the currentLiveRange (the second number in
+         * the copy range array), use endCurrentChunk to:
+         *   - stop adding lines to the current chunk
+         *   - add that complete chunk to the allChunks array as a live chunk
+         *   - start a fresh currentChunk array
+         */
         endCurrentChunk({ live: true });
+        /**
+         * Then look for the next copy range. If none exists, currentLiveRange
+         * will be undefined.
+         */
         currentLiveRange = mutableCopyRanges && mutableCopyRanges.shift();
       }
+      /**
+       * Push the contents of the line you're currently iterating over
+       * to the currentChunk.
+       */
       currentChunk.push({
         highlighted: chunk,
         raw: rawCodeLines[i]
       });
     }
+
+    /**
+     * After iterating through all lines, if the currentChunk still contains
+     * contents that means the chunk has not yet been added to the allChunks
+     * array (which is doen with endCurrentChunk).
+     */
     if (currentChunk.length) {
-      endCurrentChunk({ live: false });
+      /**
+       * If a currentLiveRange also still exists after iterating through
+       * all lines, that means the live chunk continues to the last line of
+       * the code snippet.
+       */
+      if (currentLiveRange) {
+        /**
+         * If a currentLiveRange exists, end the current chunk and add it to
+         * the allChunks array as a live chunk.
+         */
+        endCurrentChunk({ live: true });
+      } else {
+        /**
+         * If a currentLiveRange does not exist, end the current chunk and add
+         * it to the allChunks array as a non-live chunk.
+         */
+        endCurrentChunk({ live: false });
+      }
```

Comment:
@katydecorah while I was writing code comments, I realized there was a cleaner way to do this and changed my approach. Can you take a second look? 

Basically, I reverted all other changes I made and handled it here. Hopefully the code comments thoroughly explain my approach, but please let me know if you have any questions!